code/addons/test/src/manager.tsx

```diff
@@ -89,62 +96,51 @@ addons.register(ADDON_ID, (api) => {
         }
 
         if (update.details?.testResults) {
-          (async () => {
-            await api.experimental_updateStatus(
-              TEST_PROVIDER_ID,
-              Object.fromEntries(
-                // @ts-expect-error: TODO: Fix types
-                update.details.testResults.flatMap((testResult) =>
-                  testResult.results
-                    .filter(({ storyId }) => storyId)
-                    .map(({ storyId, status, testRunId, ...rest }) => [
-                      storyId,
-                      {
-                        title: 'Component tests',
-                        status: statusMap[status],
-                        description:
-                          'failureMessages' in rest && rest.failureMessages
-                            ? rest.failureMessages.join('\n')
-                            : '',
-                        data: { testRunId },
-                        onClick: openTestsPanel,
-                        sidebarContextMenu: false,
-                      } satisfies API_StatusObject,
-                    ])
-                )
-              )
-            );
-
-            await api.experimental_updateStatus(
-              'storybook/addon-a11y/test-provider',
-              Object.fromEntries(
-                // @ts-expect-error: TODO: Fix types
-                update.details.testResults.flatMap((testResult) =>
-                  testResult.results
-                    .filter(({ storyId }) => storyId)
-                    .map(({ storyId, testRunId, reports }) => {
-                      const a11yReport = reports.find((r: any) => r.type === 'a11y');
-                      return [
-                        storyId,
-                        a11yReport
-                          ? ({
-                              title: 'Accessibility tests',
-                              description: '',
-                              status: statusMap[a11yReport.status],
-                              data: { testRunId },
-                              onClick: () => {
-                                api.setSelectedPanel('storybook/a11y/panel');
-                                api.togglePanel(true);
-                              },
-                              sidebarContextMenu: false,
-                            } satisfies API_StatusObject)
-                          : null,
-                      ];
-                    })
-                )
-              )
-            );
-          })();
+          componentTestStatusStore.set(
```

Comment:
from an addon's perspective, this change is the main difference between the old and the new API.